bootstrap/artifacts.go

```diff
@@ -0,0 +1,98 @@
+package bootstrap
+
+import (
+	"context"
+
+	"github.com/buildkite/agent/v3/tracetools"
+)
+
+func (b *Bootstrap) artifactPhase(ctx context.Context) error {
```

Comment:
A named return arg can help simplify some of the code in this function:

`func (b *Bootstrap) artifactPhase(ctx context.Context) (err error) {
...snip...

    // note this next line sees the returned value, even though it is never assigned with =
    defer func() { span.FinishWithError(err) }() 

    if err := b.preArtifactHooks(ctx); err != nil {
        return err
    }

 	if err := b.uploadArtifacts(ctx); err != nil {
 		return err
 	}

 	return b.postArtifactHooks(ctx)
}`

Reply:
Oh, feel free to ignore me - splitting the existing code should come before cleanups.